### PR TITLE
Event progress update uses userAchievements

### LINKED ISSUE #121: Event rankings and profiles are no longer pulling achievements from userAchievements
User achievements should be pulled from the classMentors/userAchievements/<profileId> node rather than userProfile.

### PR DESCRIPTION
**Pull request by dinoboff**
fixes #121

TODO:
- [x] update datastore service
- [x] fix ranking tab

### CODE REVIEW

File: src/classmentors/services.js [hunk 1]
```diff
@@ -1349,11 +1350,11 @@ export function clmDataStoreFactory(
...
-        var ranking = {
-          singPath: clmDataStore.events._solvedProblems(data.singPath),
-          codeCombat: data.badges.codeCombat.length,
-          codeSchool: data.badges.codeSchool.length
-        };
+        var ranking = {singPath: clmDataStore.events._solvedProblems(data.singPath)};
+
+        Object.keys(data.userAchievements || {}).forEach(serviceId => {
+          ranking[serviceId] = data.userAchievements[serviceId].totalAchievements || 0;
+        });
```
Comment (dinoboff) [inline src/classmentors/services.js]: Just realised ranking is not pre-calculated anymore. I will just remove that part.

**Comment by dinoboff** [comment]
@SingaporeClouds Event view and services are updated to use userAchievements.

Note that sorting still use angular orderBy filter but happens in the controller so that sorting only happens when sorting key/order changes, not for every scope digest cycle.

Things that didn't seems to be implemented (and are not implemented in this PR either):
- update ranking on participants profile/userAchievements update.
- rank in school for school event.
- ranked service (a way to edit the list of service to include in the ranking).
- paging.

Did I miss any of them? Should I implement them in this PR?

**Comment by dinoboff** [comment]
@SingaporeClouds rebased and removed left over testing code (I was arbitrary changing total count to have something to sort).
@shiquan-lim Please review and check if I missed a feature that currently supported. Also, are there other views needing updated

**Comment by SingaporeClouds** [comment]
The event rankings appear to break in a few places with this PR. There are numbers being concatenated as strings rather than added as integers and there are some "<!" symbols appearing that do not happen prior to this PR. 

Also, there appears to be a mixture of white and grey on the profile pages that I am not sure was intentional or not. Pictures attached. 

image

image

**Comment by dinoboff** [comment]
I will fix those.

There's also an issue with the refresh all button that I will fix too.

For the profile page, it's unrelated. I will fix it in an other PR.

**Comment by dinoboff** [comment]
Rebased

@SingaporeClouds Can you please review again.

**Pull request merged**